XenModel/Alerts/PolicyAlert.cs

```diff
@@ -43,16 +45,51 @@ public class PolicyAlert : IEquatable<PolicyAlert>
         public readonly string Type;
         public readonly string Text;
         public readonly DateTime Time;
+        public readonly string PolicyType;
+        public readonly int numberOfVMsFailed;
 
-        public PolicyAlert(long priority, string name, DateTime _time)
+        public PolicyAlert(long priority, string name, DateTime _time, string body, string policyName)
         {
             Type = (priority == 4 ? "info": "error");
             Time = _time;
-            Text = Message.FriendlyBody(name);
+            PolicyType = "VMSS";
+
+            if(Type == "info")
+            {
+                Text = Message.FriendlyBody(name);
+                return;
+            }
+
+            /* We reach here when message type is an error hence we need 
+             * to parse the message body accordingly */
+
+            numberOfVMsFailed = Regex.Matches(body, "VM:").Count;
+            string[] messageHeader = Regex.Split(body.Replace("\n", " "), "Details:");
+            Text = string.Format(Message.FriendlyBody(name), policyName, numberOfVMsFailed, "\n\n" + Message.FriendlyBody("vmss_details") + "\n");
+
+             /* get a list of all VMs that have encountered an error */
+            string[] vmList = Regex.Split(messageHeader[1], "VM:");
+
+            /* for each VM entry parse the data and get VM name and corresponding error message */
+            for (int vmIterator = 1; vmIterator < vmList.Length; vmIterator++)
+            {
+                vmList[vmIterator].Trim();
+                string[] tmp = Regex.Split(vmList[vmIterator], "Error:");
+                string vmName = Regex.Split(tmp[0], "UUID:")[0];
+              
+                string[] errorCode = Regex.Split(tmp[1].Replace("[", "").Replace("],", "").Replace("]", ""), "\',");
+                for (int errorCodeIterator = 0; errorCodeIterator < errorCode.Length; errorCodeIterator++)
+                    errorCode[errorCodeIterator] = errorCode[errorCodeIterator].Replace("\'", "").Trim();
+
+                string errorMessage = new Failure(errorCode).ShortMessage;
+                Text += vmIterator +") " + vmName.Trim() + ": " + errorMessage + '\n';

```

Comment:
This should be done with a string format method, for example (using a StringBuilder, as suggested above):
`stringBuilder.AppendFormat(Messages.VMSS_ALERT_VM_ERROR_FORMAT, vmIterator, vmName.Trim(), errorMessage);` where Messages.VMSS_ALERT_VM_ERROR_FORMAT is `"{0}. {1}: {2}"` Here I also suggested "." instead of ")" for the numbered list.